clippy_lints/src/write.rs

```diff
@@ -187,10 +192,31 @@ declare_lint_pass!(Write => [
 ]);
 
 impl EarlyLintPass for Write {
+    fn check_item(&mut self, _: &EarlyContext<'_>, item: &Item) {
+        if let ItemKind::Impl(_, _, _, _, Some(trait_ref), _, _) = &item.kind {
+            let trait_name = trait_ref
+                .path
+                .segments
+                .iter()
+                .last()
+                .expect("path has at least one segment")
+                .ident
+                .name
+                .as_str();
+            if trait_name == "Debug" {
```

Comment:
This will also stop linting in user defined traits, named `Debug`. But I guess this is a FN, we can live with.